src/Microsoft.TestPlatform.CrossPlatEngine/Execution/ExecutionManager.cs

```diff
@@ -88,6 +93,8 @@ public void Initialize(IEnumerable<string> pathToAdditionalExtensions)
             ITestCaseEventsHandler testCaseEventsHandler,
             ITestRunEventsHandler runEventsHandler)
         {
+            //unsubscrive session logger
+            this.sessionMessageLogger.TestRunMessage -= this.TestSessionMessageHandler;
```

Comment:
this should be done in Initialize itself

Reply:
Moved in to the Initialize()